File: abc_bank/src/main/java/com/abc_bank/abc_bank/accounts/entity/Account.java
package com.abc_bank.abc_bank.accounts.entity;


import com.abc_bank.abc_bank.auth_users.entity.User;
import com.abc_bank.abc_bank.enums.AccountStatus;
import com.abc_bank.abc_bank.enums.AccountType;
import com.abc_bank.abc_bank.enums.Currency;
import com.abc_bank.abc_bank.transaction.entity.Transaction;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@Builder
@Table(name="accounts")
@AllArgsConstructor
@NoArgsConstructor
public class Account {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;

    @Column(nullable=false,unique=true,length=15)
    private String accountNumber;

    private BigDecimal BALANCE=BigDecimal.ZERO;

    @Enumerated(EnumType.STRING)
    @Column(nullable=false)
    private AccountType accountType;

    @ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name="user_id", nullable=false)
    private User user;

    @Enumerated(EnumType.STRING)
    private Currency currency;

    @Enumerated(EnumType.STRING)
    private AccountStatus status;

    @OneToMany(mappedBy="account",cascade=CascadeType.ALL,orphanRemoval = true,fetch=FetchType.LAZY)
    private List<Transaction> transactions=new ArrayList<>();


    private LocalDateTime closedAt;
    private LocalDateTime createdAt=LocalDateTime.now();
    private LocalDateTime updatedAt;
}

File: abc_bank_py/app/models/account.py
from datetime import datetime
from decimal import Decimal
from typing import TYPE_CHECKING

from sqlalchemy import DateTime, Enum as SAEnum, ForeignKey, Numeric, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db import Base
from app.enums import AccountStatus, AccountType, Currency

if TYPE_CHECKING:
    from app.models.transaction import Transaction
    from app.models.user import User


class Account(Base):
    __tablename__ = "accounts"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    account_number: Mapped[str] = mapped_column(String(15), unique=True, nullable=False, index=True)
    balance: Mapped[Decimal] = mapped_column(Numeric(18, 2), nullable=False, default=Decimal("0"))
    account_type: Mapped[AccountType] = mapped_column(SAEnum(AccountType), nullable=False)
    currency: Mapped[Currency] = mapped_column(SAEnum(Currency), nullable=False)
    status: Mapped[AccountStatus] = mapped_column(
        SAEnum(AccountStatus), nullable=False, default=AccountStatus.ACTIVE
    )
    closed_at: Mapped[datetime | None] = mapped_column(DateTime)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False
    )

    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False, index=True)
    user: Mapped["User"] = relationship(back_populates="accounts")
    transactions: Mapped[list["Transaction"]] = relationship(
        back_populates="account", cascade="all, delete-orphan"
    )
